src/panel/direction/DirectionPanel.jsx

```diff
@@ -0,0 +1,226 @@
+/* globals _ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import Panel from 'src/components/ui/Panel';
+import DirectionForm from './DirectionForm';
+import RouteResult from './RouteResult';
+import DirectionApi, { modes } from 'src/adapters/direction_api';
+import Telemetry from 'src/libs/telemetry';
+import nconf from '@qwant/nconf-getter';
+import LatLonPoi from 'src/adapters/poi/latlon_poi';
+import Device from 'src/libs/device';
+import Error from 'src/adapters/error';
+
+const poiUrlValue = poi =>
+  typeof poi.toUrl === 'function' ? poi.toUrl() : `${poi.id}@${poi.name}`;
+
+export default class DirectionPanel extends React.Component {
+  static propTypes = {
+    origin: PropTypes.string,
+    destination: PropTypes.string,
+    poi: PropTypes.object,
+    mode: PropTypes.string,
+  }
+
+  constructor(props) {
+    super(props);
+
+    Telemetry.add(Telemetry.ITINERARY_OPEN, null, null,
+      props.poi && Telemetry.buildInteractionData({
+        id: props.poi.id,
+        source: props.poi.meta ? props.poi.meta.source : props.poi.name,
+        template: 'simple',
+        zone: 'detail',
+        element: 'itinerary',
+      })
+    );
+
+    this.vehicles = [modes.DRIVING, modes.WALKING, modes.CYCLING];
+    if (nconf.get().direction.publicTransport.enabled) {
+      this.vehicles.splice(1, 0, modes.PUBLIC_TRANSPORT);
+    }
+
+    const activeVehicle = this.vehicles.indexOf(props.mode) !== -1
+      ? props.mode : modes.DRIVING;
+
+    this.state = {
+      vehicle: activeVehicle,
+      origin: null,
+      destination: props.poi || null,
```

Comment:
Oh no, that feature seems so counter-intuitive to me that I overlooked it :(… Indeed it would require a mechanism to move that upstream, but it should be easy enough. Thanks for noticing, I'll rework it.

Reply:
Done in https://github.com/QwantResearch/erdapfel/pull/479/commits/07843012093c9db28e5644a6a8b797a8ec9e9e8a
I also reworked a bit the initialization code, so a bit easier to read and accomodate all cases.